### DISCUSSION TITLE
Specific GraphQLError when handling exceptions in a SpringGraphQLContextFactory

### DISCUSSION THREAD

**Discussion by j-davies** [ExpediaGroup/graphql-kotlin#1027] [Q&A]
Hello,
I have a `SpringGraphQLContextFactory` that decodes a jwt token, at the moment if the token is invalid a 500 error is returned to the client.

I would like to send a GraphQLError `AuthenticationError` with code `UNAUTHENTICATED` in a similar way to [Apollo Server](https://www.apollographql.com/docs/apollo-server/data/errors/#codes) allows.

I cannot send a GraphQLError directly from the context, because it does not have access to the response. 
I have looked at using at `DataFetcherExceptionHandler`, but that is only invoked on exceptions that happen during the graphql request excution.  
Also, I looked at creating my own `SpringGraphQLServer` and handling specific exceptions with a custom error, but this is a bit messy. 
A specific error could be returned if moved the token validation into the Query/Mutation, but I would like to have it all handled transparently in the Context.

What would be your recommended approach?

**Comment by dariuszkuc** [comment]
Hello 👋 
With the recent server refactor (https://github.com/ExpediaGroup/graphql-kotlin/pull/988) you could create custom instance of `GraphQLServer` that would handle the exception thrown directly from the [SpringGraphQLContextFactory](https://github.com/ExpediaGroup/graphql-kotlin/blob/master/servers/graphql-kotlin-spring-server/src/main/kotlin/com/expediagroup/graphql/server/spring/execution/SpringGraphQLContextFactory.kt#L27) and return appropriate response. This is available in version `4.0.0-alpha.11`+. See [wiki](https://expediagroup.github.io/graphql-kotlin/docs/server/graphql-server) for details.

In older versions you would need to create your own custom [ContextWebFilter](https://github.com/ExpediaGroup/graphql-kotlin/blob/4.0.0-alpha.10/graphql-kotlin-spring-server/src/main/kotlin/com/expediagroup/graphql/spring/GraphQLExecutionConfiguration.kt#L65) that would send the error on the `ServerWebExchange`.
